src/org/zaproxy/zap/extension/api/API.java

```diff
@@ -239,8 +239,14 @@ public boolean handleApiRequest (HttpRequestHeader requestHeader, HttpInputStrea
 			JSONObject params = getParams(requestHeader.getURI().getEscapedQuery());
 
 			if (shortcutImpl != null) {
+				if (!getOptionsParamApi().isDisableKey() && !getOptionsParamApi().isNoKeyForSafeOps()) {
+					if ( ! this.hasValidKey(requestHeader, params)) {
+						throw new ApiException(ApiException.Type.BAD_API_KEY);
```

Comment:
Isn't a break or return needed here? It seems the shortcut will still be processed after throwing the exception unless I'm missing something...doesn't the current flow count on the exception being unhandled?